src/screens/Hooks/OMPSwitchHooks.res

```diff
@@ -158,9 +157,33 @@ let useInternalSwitch = () => {
     } catch {
     | Exn.Error(e) => {
         let err = Exn.message(e)->Option.getOr("Failed to switch!")
-        showToast(~message=err, ~toastType=ToastError)
         Exn.raiseError(err)
       }
     }
   }
 }
+
+let useOMPData = () => {
+  open OMPSwitchUtils
+  let merchantList = Recoil.useRecoilValueFromAtom(HyperswitchAtom.merchantListAtom)
+  let orgList = Recoil.useRecoilValueFromAtom(HyperswitchAtom.orgListAtom)
+  let profileList = Recoil.useRecoilValueFromAtom(HyperswitchAtom.profileListAtom)
+  let {userInfo} = React.useContext(UserInfoProvider.defaultContext)
+  let getList: unit => OMPSwitchTypes.ompList = () => {
+    {
+      orgList,
+      merchantList,
+      profileList,
+    }
+  }
+
+  let getNameForId = entityType =>
+    switch entityType {
+    | #Organization => currentOMPName(orgList, userInfo.orgId)
+    | #Merchant => currentOMPName(merchantList, userInfo.merchantId)
+    | #Profile => currentOMPName(profileList, userInfo.profileId)
+    | _ => ""
+    }
+
+  (getList, getNameForId)
```

Comment:
can you return an object else everywhere you need to destructure whole tuple.
And will somehow look like this when other is not needed
```js
let (_, getNameForId) = OMPSwitchHooks.useOMPData()
```

Reply:
As we are only functions from the looks we can use tuple as of now and when the use-case will come for extending it will introduce a type and return object from the hook